optaplanner-core/src/main/java/org/optaplanner/core/config/solver/SolverConfig.java

```diff
@@ -235,7 +235,11 @@ public static SolverConfig createFromXmlReader(Reader reader, ClassLoader classL
     protected Integer moveThreadBufferSize = null;
     protected Class<? extends ThreadFactory> threadFactoryClass = null;
 
+    /**
+     * @deprecated This functionality will be removed in the next major version of OptaPlanner.
```

Comment:
deprecated in favor of the Quarkus extension or Spring Boot starter which handle annotation scanning more efficiently.